lib/bus.js

```diff
@@ -104,19 +106,19 @@ Bus.prototype.send = function(target, code, data, callback) {
  * @return {Command}
  */
 Bus.prototype.parse = function(message) {
-  if (!isValid(message) || source(message) !== this.gateway) return;
+  if (!isValid(message) || (!this.broadcast && (source(message) !== this.gateway))) return;
+  if (!isValid(message)) return;
```

Comment:
Also, why bus working in broadcast mode should parse messages from any gateway? I think its a separate concept. If you need to interact with another gateway, just create separate Bus instance.

I know there are an issue with that: UDP port using by Bus object will be the same as gateway port so it's not possible to have two separate Bus instances if both gateways have same port.

It will be better to this fix issue by selecting any free UDP port instead of using `broadcast` property to intercept messages from other gateways.

Reply:
If you put the gateways on different ports they can no longer communicate between eachother - they do this over UDP/6000.  ie - when using a layer 2 ethernet connection to bridge two HDL networks via IP interfaces - all comms need to be over 6000 (as I understand it).

Agree the second validity check is in error.

Reply:
Listening port of the gateway itself could be changed with HDL software. But I'm speaking about UDP port `Bus` instance using locally. Current version of `smart-bus` mistakenly use gateway listening port for local UDP socket. UDP port on localhost could be selected randomly. 

I can try to fix this issue on this weekend.

Reply:
I don't think you can - you need to bind to UDP/6000 to listen to messages from the gateway - agree the send port can be random, but the problem is the binding/receive port anyway.  The gateway transmits and received on port 6000 and I think changing this will break things.

Perhaps an option might be to seperate the UDP socket from the bus object, and have the bus objects hook into the socket via events, thus sharing UDP/6000?

Reply:
For more information see this https://github.com/caligo-mentis/smart-bus/issues/4#issuecomment-411653349

Reply:
Yes that's talking about similar issues - there's no point in opening up other ports for other bus', as they all broadcast on UDP/6000 (this is how they can talk to eachother), so it's important to always listen on UDP/6000.

Reply:
Yes, but port for local UDP socket is not important. This library must allow to do following:

```js
var nodejsGateway = new SmartBus('hdl://1.50@192.168.1.250:6000');
var gatewayForEnviro = new SmartBus('hdl://253.254@192.168.1.250:6000'); 
```

and now it's impossible due to mistakenly chosen port on socket initialization.

Reply:
No but it is important - because all traffic from physical IP gateways will arrive on UDP/6000.  I think, if you want to instantiate multiple bus interfaces like that, you need to do something like this (sharing the socket):
```js
var network = new Network('192.168.1.250:6000');
var nodejsGateway = new SmartBus('1.50', network);
var gatewayForEnviro = new SmartBus('1.50', network);
```

Reply:
Yes, this will be possible if `Gateway` would be extracted from `Bus` class:

```js
var bus = new SmartBus('192.168.1.255:6000');

var nodejsGateway = bus.gateway('1.50');
var gatewayForEnviro = bus.gateway('253.254');
```

Reply:
Yes! I think that is the way forward to achieve the functionality you're wanting.  I've just uploaded a couple of commits including a `sendAs` method that allows the specifying of the sender device as an interim workaround.  And updated the readme as requested.

Reply:
Can we accept my sendAs function for now, then create a new fork or branch to work on the separation of function?  See latest commit.

Reply:
Sure, just add tests for new features and I will merge this PR